### PR TITLE
Polish ANSI formatting and underlying code

### PR DESCRIPTION
**Pull request by translatenix**
- change line numbers from blue to faint to improve legibility on dark backgrounds
- use EnumSet throughout
- move all ANSI classes into package core.util (no need to split them over util and runtime)
- rename AnsiCodingStringBuilder to AnsiStringBuilder

### CODE REVIEW

File: pkl-core/src/main/java/org/pkl/core/util/AnsiTheme.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ private AnsiTheme() {}
 
   public static final AnsiCode STACK_FRAME = AnsiCode.FAINT;
   public static final AnsiCode STACK_TRACE_MARGIN = AnsiCode.YELLOW;
-  public static final AnsiCode STACK_TRACE_LINE_NUMBER = AnsiCode.BLUE;
+  public static final AnsiCode STACK_TRACE_LINE_NUMBER = AnsiCode.BRIGHT_BLUE;
```
Comment (bioball) [inline pkl-core/src/main/java/org/pkl/core/util/AnsiTheme.java:31]: How about `AnsiCode.FAINT`? The bright blue draws attention to the line number, but it's arguably one of the less important parts of the error message.

Screenshot 2024-11-04 at 4 31 06 PM

Comment (translatenix) [reply pkl-core/src/main/java/org/pkl/core/util/AnsiTheme.java]: Looks even better (updated).

By the way, is there a reason why I can't set colors in `settings.pkl`?

Comment (bioball) [reply pkl-core/src/main/java/org/pkl/core/util/AnsiTheme.java]: I haven't put too much thought into it so far. I don't see any reason why it shouldn't be there, though.

**Review by bioball — APPROVED** [review]
LGTM; looking for one more thumb from either @holzensp or @stackoverflow

**Review by stackoverflow — APPROVED** [review]
LGTM!

**Pull request merged**